src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -179,84 +175,81 @@ export class StudyViewPageStore {
 
     @computed
     get mutationProfileIds() {
         return this.molecularProfiles
             .result
-            .filter(profile=> profile.molecularAlterationType === "MUTATION_EXTENDED")
-            .map(profile=>profile.molecularProfileId);
+            .filter(profile => profile.molecularAlterationType === "MUTATION_EXTENDED")
+            .map(profile => profile.molecularProfileId);
     }
 
     @computed
     get cnaProfileIds() {
         return this.molecularProfiles
             .result
-            .filter(profile=> profile.molecularAlterationType === "COPY_NUMBER_ALTERATION" && profile.datatype === "DISCRETE")
-            .map(profile=>profile.molecularProfileId);
+            .filter(profile => profile.molecularAlterationType === "COPY_NUMBER_ALTERATION" && profile.datatype === "DISCRETE")
+            .map(profile => profile.molecularProfileId);
     }
 
     readonly clinicalAttributes = remoteData({
-        invoke: () => {
-            return defaultClient.getAllClinicalAttributesInStudyUsingGET({
-                studyId: this.studyId
-            })
-        },
+        await: () => [this.studies],
+        invoke: () => defaultClient.fetchClinicalAttributesUsingPOST({
+            studyIds: this.studies.result.map(study => study.studyId)
+        }),
         default: [],
-        onResult:(attributes:ClinicalAttribute[])=>{
-            attributes.forEach(attribute=>{
-                this._attributeChartVisibility.set(attribute.clinicalAttributeId,false);
-            })
+        onResult: (attributes) => {
+            let newMap: { [id: string]: ChartMeta } = _.reduce(attributes, (acc: { [id: string]: ChartMeta }, attribute) => {
+                const clinicalDataType: ClinicalDataType = attribute.patientAttribute ? 'PATIENT' : 'SAMPLE';
+                const uniqueKey = clinicalDataType + '_' + attribute.clinicalAttributeId;
+                //TODO: currently only piechart is handled
+                if (attribute.datatype === 'STRING') {
+                    acc[uniqueKey] = { clinicalAttribute: attribute, uniqueKey: uniqueKey, defaultChartType: ChartType.PIE_CHART };
+                }
+                return acc
+            }, {});
+            this._clinicalAttributesMetaSet = newMap;
         }
     });
 
-    readonly clinicalAttributeSet = remoteData<{[id:string]:ClinicalAttribute}>({
-        await: () => [this.clinicalAttributes],
-        invoke: () => Promise.resolve(_.keyBy(this.clinicalAttributes.result, x=>x.clinicalAttributeId))
-    },{});
-
-    @computed get visibleAttributes(){
-        const attributes = this.clinicalAttributeSet.result
-        return _.reduce(this._attributeChartVisibility.toJS(),(acc:ClinicalAttribute[],next,key)=>{
-            if(next && attributes[key]){
-                acc.push(attributes[key])
-            }
-            return acc;
-        },[])
+    @computed get visibleAttributes(): ChartMeta[] {
+        if (this.initalClinicalDataCounts.isComplete) {
```

Comment:
loading state should not handled inside of @computed getter.    this hides the loading state from calling code, which only knows whether array is empty or not (could be empty for other reasons).   Conditional should be in calling code.